viz_scripts/scaffolding.py

```diff
@@ -125,18 +127,44 @@ def load_viz_notebook_data(year, month, program, study_type, dic_re, dic_pur=Non
     if "distance" in expanded_ct.columns:
         unit_conversions(expanded_ct)
 
-    # Mapping new mode labels with dictionaries
-    # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
-    if "mode_confirm" in expanded_ct.columns:
-        expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_re)
-    if study_type == 'program':
+    # Extract translations_data for dynamic_labels
+    if has_dynamic_labels:
+        print("This json has dynamic label")
+        req = requests.get(dynamic_labels_url)
+        if req.status_code != 200:
+            print("Unable to download dynamic_labels")
+        else:
+            print("dynamic labels download was successful.")
+            dynamic_labels = json.loads(req.text)
+
+        # Extract translations section
+        translations_data = dynamic_labels["translations"]["en"]
+
+        # Map new mode labels with translations_data from dynamic_labels
         # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
-        if 'replaced_mode' in expanded_ct.columns:
-            expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(dic_re)
+        if "mode_confirm" in expanded_ct.columns:
+            expanded_ct["Mode_confirm"] = expanded_ct["mode_confirm"].map(translations_data)
+        if study_type == 'program':
+            # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
+            if 'replaced_mode' in expanded_ct.columns:
+                expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(translations_data)
+            else:
+                print("This is a program, but no replaced modes found. Likely cold start case. Ignoring replaced mode mapping")
         else:
-            print("This is a program, but no replaced modes found. Likely cold start case. Ignoring replaced mode mapping")
+                print("This is a study, not expecting any replaced modes.")
     else:
-            print("This is a study, not expecting any replaced modes.")
+        # Mapping new mode labels with dictionaries
+        # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
+        if "mode_confirm" in expanded_ct.columns:
+            expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_re)
```

Comment:
here. It seems like it can be unified into a single call that performs the map with the appropriate input

Reply:
Yes, I have introduced a new function call update_expaned_ct(expanded_ct, new_mode, old_mode, dic_mapping).